File: chat-application-backend-go/routes/chatRoutes.go
// routes/chatRoutes.go
package routes

import (
	"net/http"

	"github.com/gorilla/mux"

	"chat-application-backend-go/controllers"
	"chat-application-backend-go/middleware"
)

func ChatRoutes(r *mux.Router) {
	r.Handle("/messages", middleware.AuthMiddleware(http.HandlerFunc(controllers.SendMessage))).Methods("POST")
	r.Handle("/messages", middleware.AuthMiddleware(http.HandlerFunc(controllers.GetMessages))).Methods("GET")
}

File: chat-application-backend-python/routes/chat_routes.py
# routes/chat_routes.py
from flask import Blueprint
from controllers.chat_controller import send_message, get_messages
from middleware.auth_middleware import auth_middleware

chat_bp = Blueprint('chat', __name__)

chat_bp.route('/messages', methods=['POST'])(auth_middleware)(send_message)
chat_bp.route('/messages', methods=['GET'])(auth_middleware)(get_messages)

